superset-frontend/src/types/ChartSource.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+export enum ChartSource {
+  explore = 'explore',
+  dashboard = 'dashboard',
+}
```

Comment:
one nite, the member of the enum is a constant so should use uppercase naming.

Reply:
Actually, Typescript recommends PascalCase for [enums](https://www.typescriptlang.org/docs/handbook/enums.html). You can check Ville's comment [here](https://github.com/apache/superset/pull/21438#discussion_r970821861).

Reply:
Updated to PascalCase.